Sources/AsyncHTTPClient/HTTPHandler.swift

```diff
@@ -68,30 +68,52 @@ extension HTTPClient {
             }
         }
 
+        public struct Length: Hashable, Sendable {
+            @usableFromInline
+            let length: RequestBodyLength
+
+            private init(_ length: RequestBodyLength) {
+                self.length = length
+            }
+
+            public static let unknown = Length(.unknown)
+
+            public static func known(_ length: Int64) -> Length {
+                return Length(.known(length))
+            }
+        }
+
         /// Body size. If nil,`Transfer-Encoding` will automatically be set to `chunked`. Otherwise a `Content-Length`
         /// header is set with the given `length`.
         @available(*, deprecated, renamed: "contentLength")
         public var length: Int? {
             get {
-                self.contentLength.flatMap { Int($0) }
+                if case .known(let count) = self.contentLength.length {
+                    return Int(count)
+                }
+                return nil
             }
             set {
-                self.contentLength = newValue.flatMap { Int64($0) }
+                if let newValue = newValue {
+                    self.contentLength = Length.known(Int64(newValue))
+                } else {
+                    self.contentLength = Length.unknown
+                }
             }
         }
 
         /// Body size. If nil,`Transfer-Encoding` will automatically be set to `chunked`. Otherwise a `Content-Length`
         /// header is set with the given `contentLength`.
-        public var contentLength: Int64?
+        public var contentLength: Length
```

Comment:
This is a new not an API break because we haven't released the API yet afaik but I agree exposing this var is weird. @fabianfett what do you think?

Reply:
We haven't released this API yet. Therefore this will not be a breaking change. Generally I'm in favor of using `@_disfavoredOverload` on the `Int` version of this API as well as the initializers that use trailing closure syntax.